### ISSUE TITLE
Solution generation fails when multiple projects include the same shared items project

### ISSUE THREAD

**Issue by OskiKervinen-MF** [microsoft/slngen#444]
Thank you for implementing #440. It works when only a single project in the dependency graph refers to the shared items project. But sadly it fails when multiple projects depend on the same shared items project. Which they usually do, that being the point of shared projects.

Example:
* SharedUser.vcxproj (exe) depends on SharedItems.vcxitems and SharedUsingLibrary.vcxproj (DLL).
* SharedUsingLibrary.vcxproj depends on SharedItems.vcxitems as well.

This results in an error like:
```
"C:\GIT\poke\CppPoking\SharedProject\SharedUser\SharedUser.vcxproj" (Restore;SlnGen target) (1) ->
(SlnGen target) ->
  SlnGen : error : System.InvalidOperationException: An equivalent project (a project with the same global properties a
nd tools version) is already present in the project collection, with the path "C:\GIT\poke\CppPoking\SharedProject\Shar
edItems\SharedItems.vcxitems". To load an equivalent into this project collection, unload this project first. [C:\GIT\p
oke\CppPoking\SharedProject\SharedUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.Build.Shared.ErrorUtilities.ThrowInvalidOperation(String resourceName, Object[] args) [
C:\GIT\poke\CppPoking\SharedProject\SharedUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.Build.Evaluation.ProjectCollection.LoadedProjectCollection.AddProject(Project project)
[C:\GIT\poke\CppPoking\SharedProject\SharedUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.Build.Evaluation.ProjectCollection.OnAfterRenameLoadedProject(String oldFullPathIfAny,
Project project) [C:\GIT\poke\CppPoking\SharedProject\SharedUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.Build.Evaluation.Project.ProjectImpl.Initialize(IDictionary`2 globalProperties, String
toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) [C:\GIT\
poke\CppPoking\SharedProject\SharedUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.Build.Evaluation.Project..ctor(String projectFile, IDictionary`2 globalProperties, Stri
ng toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings, Evalu
ationContext evaluationContext, IDirectoryCacheFactory directoryCacheFactory) [C:\GIT\poke\CppPoking\SharedProject\Shar
edUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.Build.Evaluation.Project.FromFile(String file, ProjectOptions options) [C:\GIT\poke\Cpp
Poking\SharedProject\SharedUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.VisualStudio.SlnGen.ProjectLoading.ProjectGraphProjectLoader.CreateProject(String proje
ctFullPath, IDictionary`2 globalProperties, ProjectCollection projectCollection) [C:\GIT\poke\CppPoking\SharedProject\S
haredUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.VisualStudio.SlnGen.ProjectLoading.ProjectGraphProjectLoader.LoadProjects(IEnumerable`1
 projectPaths, ProjectCollection projectCollection, IDictionary`2 globalProperties) [C:\GIT\poke\CppPoking\SharedProjec
t\SharedUser\SharedUser.vcxproj]
  SlnGen : error : at Microsoft.VisualStudio.SlnGen.ProjectLoading.ProjectLoader.LoadProjects(FileInfo msbuildExeFileIn
fo, ProjectCollection projectCollection, IEnumerable`1 entryProjects, IDictionary`2 globalProperties, ISlnGenLogger log
ger) [C:\GIT\poke\CppPoking\SharedProject\SharedUser\SharedUser.vcxproj]
```

**Comment by michael-hawker** [comment]
FYI, didn't notice this with our C# based projects; so this may be specific to C++ usage.

@OskiKervinen-MF can you create a minimal SLN which shows it working and then where slngen fails? I know that'll help the slgnen team debug.

**Comment by OskiKervinen-MF** [comment]
Sure thing, here: 
[SharedProject.zip](https://github.com/microsoft/slngen/files/10516246/SharedProject.zip)

Calling `msbuild .\SharedUser.vcxproj /t:"Restore;SlnGen" /p:SlnGenLaunchVisualStudio=false` produces the error.

**Comment by OskiKervinen-MF** [comment]
I have also verified that the sample solution in the message above work if the fix in #445 is applied.
